snapcraft/internal/meta/_desktop.py

```diff
@@ -45,11 +45,11 @@ def parse_and_reformat(self):
         self._parser.read(self._path, encoding='utf-8')
         section = 'Desktop Entry'
         if section not in self._parser.sections():
-            raise errors.InvalidDesktopFileError(
-                self._filename, "missing 'Desktop Entry' section")
+            raise errors.MissingDesktopSectionError(
+                self._filename, "missing 'Desktop Entry' section. \n Check in home directory that .desktop file exists ")
         if 'Exec' not in self._parser[section]:
-            raise errors.InvalidDesktopFileError(
-                self._filename, "missing 'Exec' key")
+            raise errors.MissingDesktopExecError(
+                self._filename, "missing 'Exec' key \n no Desktop file-Check if file exists or not.")
```

Comment:
Passing a message in here is no longer necessary. Just pass the filename and do the rest in the exception itself.